README.md

```diff
@@ -476,37 +608,26 @@ candidates_pd = \
                      number_test_rows=20)
 ```
 
-Note that the only difference compared to the previous command is `ranking_metric="conf"`.
 
-Similarly via command line:
+Note that the only difference compared to the previous command is `ranking_metric="cosine"`, or via command line:
 
 ```bash
-DeezyMatch --deezy_mode candidate_ranker -comb ./combined/test -rm conf -t 0.51 -n 1 -sz 4 -o test_candidates_deezymatch -mp ./models/finetuned_test001/finetuned_test001.model -v ./models/finetuned_test001/finetuned_test001.vocab -tn 20
+DeezyMatch --deezy_mode candidate_ranker -comb ./combined/test -rm cosine -t 0.51 -n 1 -sz 4 -o test_candidates_deezymatch -mp ./models/finetuned_test001/finetuned_test001.model -v ./models/finetuned_test001/finetuned_test001.vocab -tn 20
 ```
 
-* Select candidates based on cosine similarity:
 
-```python
-from DeezyMatch import candidate_ranker
+## Tips / Suggestions on DeezyMatch functionalities
 
-# Find candidates
-candidates_pd = \
-    candidate_ranker(scenario="./combined/test/", 
-                     ranking_metric="cosine", 
-                     selection_threshold=0.51, 
-                     num_candidates=1, 
-                     search_size=4, 
-                     output_filename="test_candidates_deezymatch", 
-                     pretrained_model_path="./models/finetuned_test001/finetuned_test001.model", 
-                     pretrained_vocab_path="./models/finetuned_test001/finetuned_test001.vocab", 
-                     number_test_rows=20)
-```
+### Candidate ranker
 
-Or via command line:
+* As already mentioned, based on our experiments, `conf` is not a good metric for ranking candidates. Consider using `faiss` or `cosine`.
+
+* Adding prefix/suffix to input strings (see `prefix_suffix` option in the input file) can greatly enhance the ranking results.
+
+* In `candidate_ranker`, the user specifies a `ranking_metric` based on which the candidates are selected. However, DeezyMatch also reports the values of other metrics for those candidates. For example, if the user selects `ranking_metric="faiss"`, the candidates are selected based on the `faiss`-distance metric. At the same time, the values of `cosine` and `conf` metrics for **those candidates** are also reported.
+
+* In most use cases, `search_size` should be set `>= num_candidates`. However, if `num_candidates` is very large, it is better to set the `search_size` to lower values. 
```

Comment:
I think it would be good to explain the rationale behind this, to help the user decide.

Reply:
@mcollardanuy I tried...please let me know if it is not clear.

Reply:
Hi @kasra, it's clear, thanks!